src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,40 +1,96 @@
-import React from 'react';
+import { useState, useEffect } from 'react';
+import { getUser } from '../../api';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions } from '../../features/currentTodo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
 
 export const TodoModal: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const remove = () => dispatch(actions.removeTodo());
+  const openedTodo = useAppSelector(state => state.currentTodo);
+  const [error, setError] = useState<string>('');
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    const loadData = async () => {
+      if (openedTodo) {
+        try {
+          const user = await getUser(openedTodo.userId);
+
+          setSelectedUser(user);
+        } catch {
+          setError('Something went wrong!');
+        }
+      } else {
+        setSelectedUser(null);
+      }
+    };
+
+    loadData();
+  }, [openedTodo]);
+
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      <Loader />
-
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
-
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
-
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">fugiat veniam minus</p>
-
-          <p className="block" data-cy="modal-user">
-            {/* For not completed */}
-            <strong className="has-text-danger">Planned</strong>
-
-            {/* For completed */}
-            <strong className="has-text-success">Done</strong>
-            {' by '}
-            <a href="mailto:Sincere@april.biz">Leanne Graham</a>
-          </p>
-        </div>
-      </div>
+      {!selectedUser && !error
```

Comment:
it's better to check the `openedTodo` instead of `selectedUser` here
```suggestion
      {!openedTodo && !error
```